table/column.go

```diff
@@ -165,7 +165,10 @@ func handleWrongUtf8Value(ctx sessionctx.Context, col *model.ColumnInfo, casted
 // CastValue casts a value based on column type.
 func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo) (casted types.Datum, err error) {
```

Comment:
Can we guarantee that this method is only called when processing `Insert` since it always set the `CastInInsert` to true before casting?

Reply:
Maybe in a update statement.